soc/espressif/common/CMakeLists.txt

```diff
@@ -41,25 +41,23 @@ endif()
 
 set_property(TARGET bintools PROPERTY disassembly_flag_inline_source)
 
-# Select the image origin depending on the boot configuration
-if(CONFIG_SOC_ESP32_APPCPU OR CONFIG_SOC_ESP32S3_APPCPU)
-  dt_nodelabel(dts_partition_path NODELABEL "slot0_appcpu_partition")
-elseif(CONFIG_SOC_ESP32C6_LPCORE)
-  dt_nodelabel(dts_partition_path NODELABEL "slot0_lpcore_partition")
-elseif(CONFIG_MCUBOOT OR CONFIG_ESP_SIMPLE_BOOT)
-  dt_nodelabel(dts_partition_path NODELABEL "boot_partition")
-else()
-  dt_nodelabel(dts_partition_path NODELABEL "slot0_partition")
+set(image_off ${CONFIG_FLASH_LOAD_OFFSET})
+
+# If CONFIG_BUILD_OUTPUT_ADJUST_LMA is defined, adjust the image offset
+# to account for the additional load memory address offset.
+# This is useful for cases where the image needs to be loaded at a specific
+# address in flash, such as when using MCUBoot and flashing secondary images.
+if(DEFINED CONFIG_BUILD_OUTPUT_ADJUST_LMA)
```

Comment:
Using the `CONFIG_BUILD_OUTPUT_ADJUST_LMA` is not the most ideal way of changing the flash offset, but since the `tests/boot/test_mcuboot` is using this, let it be here.